packit_service/service/api/projects.py

```diff
@@ -0,0 +1,152 @@
+from http import HTTPStatus
+from logging import getLogger
+from flask import make_response
+from json import dumps
+
+try:
+    from flask_restx import Namespace, Resource
+except ModuleNotFoundError:
+    from flask_restplus import Namespace, Resource
+
+from packit_service.service.api.parsers import indices, pagination_arguments
+from packit_service.models import GitProjectModel
+
+logger = getLogger("packit_service")
+
+ns = Namespace(
+    "projects", description="Repositories which have Packit Service enabled."
+)
+
+
+@ns.route("")
+class ProjectsList(Resource):
+    @ns.expect(pagination_arguments)
+    @ns.response(HTTPStatus.PARTIAL_CONTENT, "Projects list follows")
+    def get(self):
+        """List all GitProjects"""
+
+        result = []
+        first, last = indices()
+
+        projects_list = GitProjectModel.get_projects(first, last)
+        for project in projects_list:
+            project_info = {
+                "namespace": project.namespace,
+                "repo_name": project.repo_name,
+                "project_url": project.project_url,
+                "prs_handled": len(project.pull_requests),
+                "branches_handled": len(project.branches),
+                "releases_handled": len(project.releases),
+                "issues_handled": len(project.issues),
+            }
+            result.append(project_info)
+
+        resp = make_response(dumps(result), HTTPStatus.PARTIAL_CONTENT)
+        resp.headers["Content-Range"] = f"git-projects {first + 1}-{last}/*"
+        resp.headers["Content-Type"] = "application/json"
+        resp.headers["Access-Control-Allow-Origin"] = "*"
+        return resp
+
+
+@ns.route("/<forge>/<namespace>/<repo_name>/prs")
+@ns.param("forge", "Git Forge")
+@ns.param("namespace", "Namespace")
+@ns.param("repo_name", "Repo Name")
+class ProjectsPRs(Resource):
+    @ns.expect(pagination_arguments)
+    @ns.response(HTTPStatus.PARTIAL_CONTENT, "Project PRs list follows")
+    def get(self, forge, namespace, repo_name):
+        """List PRs """
+
+        result = []
+        first, last = indices()
+
+        pr_list = GitProjectModel.get_project_prs(
+            first, last, forge, namespace, repo_name
+        )
+        if not pr_list:
+            return ("", HTTPStatus.NO_CONTENT)
```

Comment:
[Reading about this](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204), I'm not sure `NO_CONTENT` is the right status here. Maybe return and empty JSON-list with `OK`?

Reply:
Sure, thanks.

Reply:
Do you want to update this?